nixos/modules/services/cluster/pacemaker/default.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.pacemaker;
+in
+{
+  # interface
+  options.services.pacemaker = {
+    enable = mkEnableOption "pacemaker";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pacemaker;
+      defaultText = literalExpression "pkgs.pacemaker";
+      description = "Package that should be used for pacemaker";
+    };
+  };
+
+  # implementation
+  config = mkIf cfg.enable {
+    assertions = [ {
+      assertion = config.services.corosync.enable;
+      message = ''
+        Enabling services.pacemaker requires a services.corosync configuration.
+      '';
+    } ];
+
+    environment.systemPackages = [ cfg.package ];
+
+    # required by pacemaker
+    users.users.hacluster = {
+      isSystemUser = true;
+      group = "pacemaker";
+      home = "/var/lib/pacemaker";
+    };
+    users.groups.pacemaker = {};
+
+    systemd.tmpfiles.rules = [
+      "d /var/lib/pacemaker 0700 hacluster pacemaker -"
+      "d /var/lib/pacemaker/cib 0700 hacluster pacemaker -"
+      "d /var/lib/pacemaker/pengine 0700 hacluster pacemaker -"
+    ];
+
+    # see ${pacemaker.src}/daemons/pacemakerd/pacemaker.service.in
+    systemd.services.pacemaker = {
+      description = "Pacemaker High Availability Cluster Manager";
+      unitConfig = {
+        Documentation = [
+          "man:pacemakerd"
+          "https://clusterlabs.org/pacemaker/doc/"
+        ];
+        ConditionKernelCommandLine = "!nocluster";
+      };
+      after = [
+        "network.target" "time-sync.target" "dbus.service"
+        "corosync.service"
+      ];
+      wants = [ "dbus.service" ];
+      requires = [ "corosync.service" ];
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+          cp -srf ${cfg.package}/var/lib/pacemaker/* /var/lib/pacemaker/
+        '';
```

Comment:
Please keep spacing consistent: 2 spaces per level of indentation.

Without looking at the contents of `${cfg.package}/var/lib/pacemaker` I'll ask: this doesn't cause any issues when run multiple times, does it?

Reply:
It's a heap of XML schema files. I'm not sure what to do. Any ideas?

Reply:
If we know these files are being edited, this cp is probably going to destroy state. If we know they aren't going to be edited, symbolic linking would be better.

Reply:
Turns out this can be dropped altogether once we no longer explicitly configure a global `--datadir`